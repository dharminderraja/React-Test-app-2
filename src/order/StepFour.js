import React, { Component } from 'react';

class StepFour extends Component {
  render() {
    const { data, submit, prevTab } = this.props;
    const { stepOne, stepTwo, stepThree } = data || {};

    return (
      <div className='step-detail'>
        <div>
          <span className='review-title'>Meal</span>
          <span className='review-value'>{stepOne ? stepOne.meal : ''}</span>
        </div>
        <div>
          <span className='review-title'>No of People</span>
          <span className='review-value'>{stepOne ? stepOne.number : ''}</span>
        </div>
        <div>
          <span className='review-title'>Restaurant</span>
          <span className='review-value'>{stepTwo ? stepTwo.restaurant : ''}</span>
        </div>
        <div>
          <span className='review-title'>Dishes</span>
          <div className='review-dishes'>
            {(stepThree ? stepThree.dishes : []).map((i, idx) => (
              <span className='dish-item' key={idx}>{`${i.dish} - ${i.number}`}</span>
            ))}
          </div>
        </div>
        <div className='nav-step'>
          <button className='prev' onClick={() => prevTab()}>Previous</button>
          <button className='next' onClick={() => submit()}>Submit</button>
        </div>
      </div>
    );
  }
}

export default StepFour;
