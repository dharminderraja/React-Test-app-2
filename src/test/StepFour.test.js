import React from 'react';
import ReactDOM from 'react-dom';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import StepFour from '../order/StepFour';

describe('Step Three', () => {
  it('should be defined', () => {
    expect(StepFour).toBeDefined();
  });

  it('should render correctly', () => {
    const component = renderer.create(<StepFour />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show values correctly', () => {
    const order = {
      stepOne: {
        meal: 'lunch',
        number: 2
      },
      stepTwo: {
        restaurant: 'Mc Donalds'
      },
      stepThree: {
        dishes: [
          {
            dish: 'Chicken Burger',
            number: 4
          }
        ]
      }
    };
    const tree = shallow(
      <StepFour
        data={order}
        prevTab={() => { }}
        submit={() => { }}
      />
    );

    expect(tree.find('.step-detail .review-value').children()).toHaveLength(3);
  });
});