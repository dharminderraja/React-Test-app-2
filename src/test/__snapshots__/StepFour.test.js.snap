// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step Three should render correctly 1`] = `
<div
  className="step-detail"
>
  <div>
    <span
      className="review-title"
    >
      Meal
    </span>
    <span
      className="review-value"
    >
      
    </span>
  </div>
  <div>
    <span
      className="review-title"
    >
      No of People
    </span>
    <span
      className="review-value"
    >
      
    </span>
  </div>
  <div>
    <span
      className="review-title"
    >
      Restaurant
    </span>
    <span
      className="review-value"
    >
      
    </span>
  </div>
  <div>
    <span
      className="review-title"
    >
      Dishes
    </span>
    <div
      className="review-dishes"
    />
  </div>
  <div
    className="nav-step"
  >
    <button
      className="prev"
      onClick={[Function]}
    >
      Previous
    </button>
    <button
      className="next"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
</div>
`;
